segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java

```diff
@@ -167,6 +171,7 @@ public ChunkedSegmentStorage(int containerId, ChunkStorage chunkStorage, ChunkMe
                 config,
                 (ScheduledExecutorService) executor);
         this.closed = new AtomicBoolean(false);
+        this.reporter = ((ScheduledExecutorService) executor).scheduleAtFixedRate(this::report, 1000, 1000, TimeUnit.MILLISECONDS);
```

Comment:
This is dangerous. You may not get a `ScheduledExecutorService`. Please fix your constructor to require that type instead.

Reply:
fixed